src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModelsDialog.java

```diff
@@ -223,14 +223,23 @@ private void onOK() {
     private PsiFile generateModelFile() {
         final NamespaceBuilder modelNamespace = getModelNamespace();
         final NamespaceBuilder resourceModelNamespace = getResourceModelNamespace();
+        final StringBuilder resourceModelFqn
+                = new StringBuilder(resourceModelNamespace.getClassFqn());
+        String resorceModelName = getResourceModelName();
+
+        if (getModelName().equals(getResourceModelName())) {
+            resourceModelFqn.append(" as ResourceModel");
```

Comment:
Can we extract the `ResourceModel` into a const and reuse it in all places we use it?